File: src/main/java/com/aau/wizard/rules/BiddingRules.java
package com.aau.wizard.rules;

import com.aau.wizard.model.Player;

import java.util.List;

public class BiddingRules {

    public static void calculateScores(List<Player> players) {
        for (Player player : players) {
            int difference = Math.abs(player.getPrediction() - player.getTricksWon());
            if (difference == 0) {
                player.setScore(player.getScore() + 20 + (10 * player.getTricksWon()));
            } else {
                player.setScore(player.getScore() - 10 * difference);
            }
        }
    }
}

File: src/main/kotlin/com/aau/wizard/core/rules/BiddingRules.kt
/*package com.aau.wizard.core.rules

import com.aau.wizard.core.model.PlayerState

object BiddingRules {
    fun calculateScores(players: List<PlayerState>) {
        for (player in players) {
            player.score += if (player.bid == player.tricksWon) {
                20 + (10 * player.bid)
            } else {
                -10 * kotlin.math.abs(player.bid - player.tricksWon)
            }
        }
    }
}
*/